main.go

```diff
@@ -472,6 +472,23 @@ func makeContextWithTrace(ctx context.Context, t transport.Transport, request *t
 
 func makeInitialRequest(out output, transport transport.Transport, serializer encoding.Serializer, req *transport.Request) {
 	response, err := makeRequestWithTracePriority(transport, req, 1)
+
+	var bs []byte
+	outSerialized := map[string]interface{}{}
+	if response != nil && len(response.Headers) > 0 {
+		outSerialized["headers"] = response.Headers
+	}
+	defer func() {
+		if len(outSerialized) == 0 {
+			return
+		}
+		bs, err = json.MarshalIndent(outSerialized, "", "  ")
+		if err != nil {
+			out.Fatalf("Failed to convert map to JSON: %v\nMap: %+v\n", err, outSerialized)
```

Comment:
Would that cause yab to quit?  Some combination of non-serializable header could break otherwise working yab.  Sould it be a non-quitting warn instead?

Reply:
This is a defer - so we still print it out last.

Moving it up here to achieve the opposite - `		out.Fatalf("Failed while making call: %s\n", buffer.String())` will cause a quit, but since we're in a defer I'm gonna get my headers out.

Reply:
We have _headers+body_ now. If _body_ marshals fine but _header_ fails, that will change the original functionality and nothing will be printed (unless I'm misreading something somewhere).

It might be better to have separate marshaling for each, and if _header_ fails skip it, printing _body_ only to preserve backward compatibility.

Reply:
Oh, I see now.

So.
1) We'll always print the whole response - it just won't be serialized. This in itself might be good enough for most cases? (most structs should be serializable?)
2) Headers are strings, so the serialization would need to fail for strings - I tried to break it (https://play.golang.org/p/7DsOl61mQRU), failed, but I guess it's possible (?)
3) Most importantly - moving headers to a separate output can be a breaking change too, right? They're currently printed as the same dict.

Unless we do something like:
```lang=go
out.Printf("{", bs)
if len(headers) {
    out.Printf(headers)
}
if len (body) {
 if len(headers) {
    out.Printf(",", bs)
 }
 out.Printf(headers)
}
out.Printf("}", bs)
```

It would be tricky to preseve the same output - we'd need a json output template of style.

Given the low likelihood of headers not being serializable, the hacks above feel like an overkill - it feels like non-serializable headers are rare enough that it shouldn't be a problem?

Also see the code above that blindly fatalfs if it fails to parse proto error details - it does not even try to handle this case, and print the output.

Happy to make a change if you find it necessary though. If so, please advise how you'd see it - it feels like we'll break some backwards compatibility either way :)